daemon/info.go

```diff
@@ -133,9 +133,13 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 		v.CPUSet = sysInfo.Cpuset
 	}
 
-	if hostname, err := os.Hostname(); err == nil {
-		v.Name = hostname
+	hostname := "<unknown>"
```

Comment:
I think this is also used by the API, and `<hostname>` is an invalid value for a "hostname". Perhaps I'm thinking to far into this, but what if a user of the API uses this value? If would be a breaking change. Perhaps we should set the "<unknown>" in the client (when *printing* the output)